From e8b83fa7c6aefdfcc372f2d1f0a18c78a1664164 Mon Sep 17 00:00:00 2001
From: Jez Austin <jezaustin@gmail.com>
Date: Mon, 22 Mar 2010 16:09:34 +0100
Subject: [PATCH 166/166] socket-client: properly handle asyncns failures

We fail to detect when people disable IPv6 in there kernels. This patch
makes sure we don't ignore this error condition.

http://pulseaudio.org/ticket/752
https://bugzilla.mozilla.org/show_bug.cgi?id=533470
http://bugs.freedesktop.org/show_bug.cgi?id=25742
---
 src/pulsecore/socket-client.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/pulsecore/socket-client.c b/src/pulsecore/socket-client.c
index b9d6950..b796cfd 100644
--- a/src/pulsecore/socket-client.c
+++ b/src/pulsecore/socket-client.c
@@ -389,7 +389,8 @@ static void asyncns_cb(pa_mainloop_api*m, pa_io_event *e, int fd, pa_io_event_fl
         goto fail;
 
     if (res->ai_addr)
-        sockaddr_prepare(c, res->ai_addr, res->ai_addrlen);
+        if (sockaddr_prepare(c, res->ai_addr, res->ai_addrlen) < 0)
+            goto fail;
 
     asyncns_freeaddrinfo(res);
 
-- 
1.7.0.5

